packages/typegpu/src/data/struct.ts

```diff
@@ -27,32 +27,107 @@ import { isDecorated, isLooseDecorated } from './attributes';
 // Public API
 // ----------
 
+/**
+ * Struct schema constructed via `d.struct` function.
+ *
+ * Responsible for handling reading and writing struct values
+ * between binary and JS representation. Takes into account
+ * the `byteAlignment` requirement of its members.
+ */
 export interface TgpuStruct<TProps extends Record<string, AnyTgpuData>>
   extends ISchema<UnwrapRecord<TProps>>,
     TgpuData<UnwrapRecord<TProps>>,
     TgpuNamable {}
 
+/**
+ * Creates a struct schema that can be used to construct gpu buffers.
+ * Describes structs with members of non-loose types.
+ * The order of members is constant, based on the order in passed in properties object.
+ *
+ * @example
+ * const CircleStruct = d.struct({ radius: d.f32, pos: d.vec3f });
+ *
+ * @param properties Record with `string` keys and `TgpuData` values,
+ * each entry describing one struct member.
+ */
 export const struct = <TProps extends Record<string, AnyTgpuData>>(
   properties: TProps,
 ): TgpuStruct<TProps> => new TgpuStructImpl(properties);
 
+/**
+ * Struct schema constructed via `d.looseStruct` function.
+ *
+ * Useful for defining tgpu vertex buffers.
+ * Members are not aligned in respect to their `byteAlignment`,
+ * unless they are explicitly decorated with the custom align attribute
+ * via `d.align` function.
+ */
 export interface TgpuLooseStruct<
   TProps extends Record<string, AnyTgpuData | AnyTgpuLooseData>,
 > extends ISchema<UnwrapRecord<TProps>>,
     TgpuLooseData<UnwrapRecord<TProps>> {}
 
+/**
+ * Creates a loose struct schema that can be used to construct tgpu vertex buffers.
+ * Describes structs with members of both loose and non-loose types.
+ *
+ * The order of members is constant, based on the order in passed in properties object.
+ * Members are not aligned in respect to their `byteAlignment`,
+ * unless they are explicitly decorated with the custom align attribute
+ * via `d.align` function.
+ *
+ * @example
+ * const CircleStruct = d.looseStruct({ radius: d.f32, pos: d.vec3f }); // packed struct with no padding
+ *
+ *  * @example
```

Comment:
Tiny nit, double asterisk.
```suggestion
 * @example
```